src/userobjects/OpenMCCellTransform.C

```diff
@@ -0,0 +1,132 @@
+                             "Type of transform to apply: 'translation' (dx,dy,dz) or 'rotation'"
+                             "(φ, θ, ψ) in degrees.");
+
+  params.addRequiredParam<std::vector<PostprocessorName>>(
+      "transform_array",
+      "An array of three postprocessors. For translation this array expects (dx, dy, dz) in mesh"
+      " units. For rotation this array expects 'φ, θ, ψ' in degrees.");
+
+  params.addClassDescription(
+      "UserObject that applies either translation or rotation on one or more OpenMC cells. "
+      "The transform is driven by a transform array of three MOOSE postprocessors");
+
+  return params;
+}
+
+OpenMCCellTransform::OpenMCCellTransform(const InputParameters & parameters)
+  : GeneralUserObject(parameters),
+    OpenMCBase(this, parameters),
+    _cell_ids(getParam<std::vector<int32_t>>("cell_ids")),
+    _transform_type(getParam<MooseEnum>("transform_type")),
+    _scaling(1.0)
+{
+  if (_cell_ids.empty())
+    paramError("cell_ids", "At least one OpenMC cell ID must be provided.");
+
+  const auto & t = getParam<std::vector<PostprocessorName>>("transform_array");
+  if (t.size() != 3)
+    paramError("transform_array",
+               "Provide exactly 3 postprocessors: 'dx dy dz' in mesh units for translation"
+               "transform or 'φ, θ, ψ' in degrees for rotation transform.");
+
+  _t0_pp = &GeneralUserObject::getPostprocessorValue("transform_array", 0);
+  _t1_pp = &GeneralUserObject::getPostprocessorValue("transform_array", 1);
+  _t2_pp = &GeneralUserObject::getPostprocessorValue("transform_array", 2);
+
+  if (_openmc_problem)
+    _scaling = _openmc_problem->scaling();
+}
+
+void
+OpenMCCellTransform::execute()
+{
+  const Real a0 = *_t0_pp;
+  const Real a1 = *_t1_pp;
+  const Real a2 = *_t2_pp;
+
+  double vec[3];
+
+  if (_transform_type == "translation")
+  {
+    vec[0] = _scaling * a0;
+    vec[1] = _scaling * a1;
+    vec[2] = _scaling * a2;
+
+    _console << "Setting OpenMC cell translations for " << _cell_ids.size() << " cell(s) to ("
+             << vec[0] << ", " << vec[1] << ", " << vec[2] << ") cm." << std::endl;
+  }
+  else if (_transform_type == "rotation")
+  {
+    vec[0] = a0;
+    vec[1] = a1;
+    vec[2] = a2;
+
+    _console << "Setting OpenMC cell rotations for " << _cell_ids.size() << " cell(s) to ("
+             << vec[0] << ", " << vec[1] << ", " << vec[2] << ") degrees." << std::endl;
+  }
+  else
+    mooseError("Unhandled transform_type: " + _transform_type);
+
+  for (const auto & cell_id : _cell_ids)
+  {
+    int32_t index = -1;
+
+    int err = openmc_get_cell_index(cell_id, &index);
+    if (err)
+      mooseError("In attempting to find OpenMC cell with ID " + std::to_string(cell_id) +
+                 ", OpenMC reported:\n\n" + std::string(openmc_err_msg));
+
+    if (_transform_type == "translation")
+      err = openmc_cell_set_translation(index, vec);
+    else
+      err = openmc_cell_set_rotation(index, vec, 3);
```

Comment:
need to technically add another `else` statement here, in case someone adds another option to the enum and forgets it here. it's tedious, but it's a not so uncommon occurence that we should protect against it everywhere.